pkg/build/admission/defaults/admission_test.go

```diff
@@ -130,3 +130,143 @@ func TestIncrementalDefaults(t *testing.T) {
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("1G"),
+				},
+			},
+			BuildResource: kapi.ResourceRequirements{
+				Requests: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("20"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("2G"),
+				},
+			},
+			ExpectedResource: kapi.ResourceRequirements{
+				Limits: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("10"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("1G"),
+				},
+				Requests: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("20"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("2G"),
+				},
+			},
+		},
+		"BuildDefaults plugin defined limits and requests ,BUILD object defined resource limits": {
+			DefaultResource: kapi.ResourceRequirements{
+				Limits: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("10"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("1G"),
+				},
+				Requests: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("10"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("1G"),
+				},
+			},
+			BuildResource: kapi.ResourceRequirements{
+				Limits: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("20"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("2G"),
+				},
+			},
+			ExpectedResource: kapi.ResourceRequirements{
+				Limits: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("20"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("2G"),
+				},
+				Requests: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("10"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("1G"),
+				},
+			},
+		},
+		"BuildDefaults plugin defined nothings ,BUILD object defined resource limits": {
+			DefaultResource: kapi.ResourceRequirements{},
+			BuildResource: kapi.ResourceRequirements{
+				Limits: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("20"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("2G"),
+				},
+				Requests: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("20"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("2G"),
+				},
+			},
+			ExpectedResource: kapi.ResourceRequirements{
+				Limits: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("20"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("2G"),
+				},
+				Requests: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("20"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("2G"),
+				},
+			},
```

Comment:
Needs a test where the build Resources object is empty to test your length checking/list initializing logic in the defaulter.